core/src/main/java/jenkins/management/ShutdownLink.java

```diff
@@ -28,38 +28,61 @@
 import hudson.model.ManagementLink;
 import hudson.security.Permission;
 import jenkins.model.Jenkins;
+import net.sf.json.JSONObject;
 import org.jenkinsci.Symbol;
+import org.kohsuke.stapler.StaplerRequest;
+import org.kohsuke.stapler.StaplerResponse;
+import org.kohsuke.stapler.verb.POST;
+
+import javax.servlet.ServletException;
+import java.io.IOException;
+import java.util.logging.Level;
+import java.util.logging.Logger;
 
 import edu.umd.cs.findbugs.annotations.NonNull;
 
-/**
- * @author <a href="mailto:nicolas.deloof@gmail.com">Nicolas De Loof</a>
- */
-@Extension(ordinal = Integer.MIN_VALUE) @Symbol("shutDown")
+@Extension(ordinal = Integer.MIN_VALUE)
+@Symbol("prepareShutdown")
 public class ShutdownLink extends ManagementLink {
 
+    private static final Logger LOGGER = Logger.getLogger(ShutdownLink.class.getName());
+
     @Override
     public String getIconFileName() {
         return "system-log-out.png";
     }
 
     public String getDisplayName() {
-        return Jenkins.get().isQuietingDown() ? Messages.ShutdownLink_DisplayName_cancel() : Messages.ShutdownLink_DisplayName_prepare();
+        return Messages.ShutdownLink_DisplayName_prepare();
     }
 
     @Override
     public String getDescription() {
-        return Jenkins.get().isQuietingDown() ? "" : Messages.ShutdownLink_Description();
+        return Messages.ShutdownLink_Description();
     }
 
     @Override
     public String getUrlName() {
-        return Jenkins.get().isQuietingDown() ? "cancelQuietDown" : "quietDown";
+        return "prepareShutdown";
     }
 
-    @Override
-    public boolean getRequiresPOST() {
-        return true;
+    @POST
+    public synchronized void doPrepare(StaplerRequest req, StaplerResponse rsp) throws IOException, ServletException, InterruptedException {
+        Jenkins.get().checkPermission(Jenkins.MANAGE);
+
+        JSONObject submittedForm = req.getSubmittedForm();
+        String inputReason = submittedForm.getString("shutdownReason");
+        String shutdownReason = inputReason.isEmpty() ? null : inputReason;
+        LOGGER.log(Level.FINE, "Shutdown requested");
```

Comment:
Might make sense to log the user here, we do this for restarts.

Reply:
- [ ] Log user